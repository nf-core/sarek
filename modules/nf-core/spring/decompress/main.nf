process SPRING_DECOMPRESS {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/spring:1.1.1--h4ac6f70_2' :
        'biocontainers/spring:1.1.1--h4ac6f70_2' }"

    input:
    tuple val(meta), path(spring)
    val(write_one_fastq_gz)

    output:
    tuple val(meta), path("*.fastq.gz"), emit: fastq
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '1.1.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    def output = write_one_fastq_gz ? "-o ${prefix}.fastq.gz" : "-o ${prefix}_R1.fastq.gz ${prefix}_R2.fastq.gz"

    """
    spring \\
        -d \\
        -g \\
        -t ${task.cpus} \\
        $args \\
        -i ${spring} \\
        ${output}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        spring: ${VERSION}
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '1.1.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    def output = write_one_fastq_gz ? "echo '' | gzip > ${prefix}.fastq.gz" : "echo '' | gzip > ${prefix}_R1.fastq.gz; echo '' | gzip > ${prefix}_R2.fastq.gz"
    """
    ${output}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        spring: ${VERSION}
    END_VERSIONS
    """
}
