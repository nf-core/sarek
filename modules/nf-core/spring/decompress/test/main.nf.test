nextflow_process {

    name "Test Process SPRING_DECOMPRESS"
    tag "modules_nfcore"
    tag "modules"
    tag "spring"
    tag "spring/decompress"
    script "../main.nf"
    process "SPRING_DECOMPRESS"



    test("Write-One-File") {

        setup {
            run("SPRING_COMPRESS") {
                script "../../compress/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        []
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = SPRING_COMPRESS.out.spring
                input[1] = true // write_one_fastq_gz
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Write-Two-Files") {

        setup {
            run("SPRING_COMPRESS") {
                script "../../compress/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test2' ], // meta map
                        file(params.modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true),
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = SPRING_COMPRESS.out.spring
                input[1] = false // write_one_fastq_gz
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Write-One-File-stub") {

        options "-stub"

        setup {
            run("SPRING_COMPRESS") {
                options "-stub"
                script "../../compress/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        []
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = SPRING_COMPRESS.out.spring
                input[1] = true // write_one_fastq_gz
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Write-Two-Files-stub") {

        options "-stub"

        setup {
            run("SPRING_COMPRESS") {
                options "-stub"
                script "../../compress/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test2' ], // meta map
                        file(params.modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + '/genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true),
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = SPRING_COMPRESS.out.spring
                input[1] = false // write_one_fastq_gz
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}