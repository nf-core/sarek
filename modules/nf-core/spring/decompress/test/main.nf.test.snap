{
    "Write-One-File stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "/home/ramprasad.neethiraj/nextflow/modules/.nf-test/tests/2a6cfab794852e23e6324eb4955668b2/work/42/aee6c82c1ca502c3b02339f597188b/test.fastq.gz"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ],
                "fastq": [
                    [
                        {
                            "id": "test"
                        },
                        "/home/ramprasad.neethiraj/nextflow/modules/.nf-test/tests/2a6cfab794852e23e6324eb4955668b2/work/42/aee6c82c1ca502c3b02339f597188b/test.fastq.gz"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-08T10:03:50.626223289"
    },
    "Write-Two-Files stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "/home/ramprasad.neethiraj/nextflow/modules/.nf-test/tests/528557b5a81e4bffb57c38b19c7aa351/work/74/fc5d116d011bcd47d6f7de8d42ac34/test2_R1.fastq.gz",
                            "/home/ramprasad.neethiraj/nextflow/modules/.nf-test/tests/528557b5a81e4bffb57c38b19c7aa351/work/74/fc5d116d011bcd47d6f7de8d42ac34/test2_R2.fastq.gz"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ],
                "fastq": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "/home/ramprasad.neethiraj/nextflow/modules/.nf-test/tests/528557b5a81e4bffb57c38b19c7aa351/work/74/fc5d116d011bcd47d6f7de8d42ac34/test2_R1.fastq.gz",
                            "/home/ramprasad.neethiraj/nextflow/modules/.nf-test/tests/528557b5a81e4bffb57c38b19c7aa351/work/74/fc5d116d011bcd47d6f7de8d42ac34/test2_R2.fastq.gz"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-08T10:03:57.417015606"
    },
    "Write-Two-Files": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "test2_R1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec",
                            "test2_R2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ],
                "fastq": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "test2_R1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec",
                            "test2_R2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T13:41:46.090761471"
    },
    "Write-One-File": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ],
                "fastq": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T13:02:07.466039653"
    },
    "Write-One-File-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ],
                "fastq": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-08T13:55:53.594615215"
    },
    "Write-Two-Files-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "test2_R1.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test2_R2.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ],
                "fastq": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "test2_R1.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test2_R2.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4711df5941f1464e3693d24dd29c705b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-08T13:56:01.212228909"
    }
}